### PR TITLE
Basic functional-based implementation of LL

### PR DESCRIPTION
**Pull request by zbrotherton**
## Justification

Addresses #3 

## Implementaion

The limelight subsystem is essentially a network table wrapper using a functional approach to distribute data, periodically updating consumers and suppliers to convey information to other subsystems. 

## Testing Done

Waiting for alphabot access

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/LimelightSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  public LimelightSubsystem(BiConsumer<Pose2d, Double> measurementConsumer) {
+    this.measurementConsumer = measurementConsumer;
+
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("stream").setNumber(1);
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(0);
+  }
+
+  public double getTagCount() {
+    return 2; // TODO: find how to pull tag count from network tables
```
[the remark this answers is not in the document]
Comment (zbrotherton) [reply src/main/java/frc/robot/subsystems/LimelightSubsystem.java]: Added a commit implementing this class

File: src/main/java/frc/robot/subsystems/LimelightSubsystem.java [hunk 2]
```diff
@@ -6,44 +6,46 @@
...
+  private Pose2d getPose2d() {
+    double[] botpose = results.targetingResults.botpose_wpiblue;
     return new Pose2d(botpose[0], botpose[1], new Rotation2d(botpose[5]));
   }
 
-  private double getTimeFromArray(double[] botpose) {
+  private double getTime() {
     // Accounts for latency
     // (https://docs.limelightvision.io/docs/docs-limelight/pipeline-apriltag/apriltag-robot-localization)
+    double[] botpose = results.targetingResults.botpose_wpiblue;
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/LimelightSubsystem.java:39]:
```suggestion
    double[] botpose = LimelightHelpers.getBotPose_wpiBlue("");
```

File: src/main/java/frc/robot/subsystems/LimelightSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  private Pose2d getPose2d() {
+    double[] botpose = results.targetingResults.botpose_wpiblue;
+    return new Pose2d(botpose[0], botpose[1], new Rotation2d(botpose[5]));
+  }
+
+  private double getTime() {
+    // Accounts for latency
+    // (https://docs.limelightvision.io/docs/docs-limelight/pipeline-apriltag/apriltag-robot-localization)
+    double[] botpose = results.targetingResults.botpose_wpiblue;
+    return Timer.getFPGATimestamp() - (botpose[6] / 1000.0);
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/LimelightSubsystem.java:40]: Its possible the returned array is not what we expect. Check the length at least before using it.

**Comment by azvanderpas** [comment]
Stale PR. Closing

**Comment by zbrotherton** [comment]
we are so back

**Comment by zbrotherton** [comment]
Testing

File: src/main/java/frc/robot/subsystems/SwerveSubsystem.java
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/SwerveSubsystem.java:1]: Telemetry.java already sets up the field and updates the robot pose. Only need the a getVisionMeasurementConsumer() addition. Revert the other changes

**Review by azvanderpas — DISMISSED** [review]

**Review by azvanderpas — APPROVED** [review]

**Pull request merged**